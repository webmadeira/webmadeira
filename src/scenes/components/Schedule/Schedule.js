import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import AppContainer from '../../../components/AppContainer/AppContainer'
import { H2 } from '../../../components/typography'
import ScheduleEntry from './components/ScheduleEntry/ScheduleEntry'

const Root = styled(AppContainer)`
  padding-top: 80px;
  padding-bottom: 80px;
  background-color: ${props => props.theme.color('secondary', 'dark')};
  color: ${props => props.theme.color('neutral', 'bright')};
`

const Title = styled(H2)`
  text-align: left;
  text-transform: uppercase;
  margin-bottom: 80px;
  
  ${({ theme }) => theme.mixins.mediaQuery.phone`
    margin-bottom: 40px;
  `};
`

const renderSchedule = schedule => schedule.map(entry => (
  <ScheduleEntry entry={entry} key={entry.id} />
))

const Schedule = ({ schedule }) => (
  <Root>
    <Title>Schedule</Title>
    <ul>
      {renderSchedule(schedule)}
    </ul>
  </Root>
)

Schedule.propTypes = {
  schedule: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
}

export default Schedule
