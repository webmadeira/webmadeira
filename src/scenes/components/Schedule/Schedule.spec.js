import React from 'react'
import { shallow } from 'enzyme'
import Schedule from './Schedule'

describe('ScheduleEntry component', () => {
  const schedule = [
    { id: 0, title: '' },
    { id: 1, title: '' },
    { id: 2, title: '' },
  ]

  it('must render properly', () => {
    const component = shallow(
      <Schedule
        schedule={schedule}
      />,
    )

    expect(component).toMatchSnapshot()
  })
})
