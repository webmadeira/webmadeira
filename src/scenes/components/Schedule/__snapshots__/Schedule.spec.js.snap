// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScheduleEntry component must render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Schedule
    schedule={
      Array [
        Object {
          "id": 0,
          "title": "",
        },
        Object {
          "id": 1,
          "title": "",
        },
        Object {
          "id": 2,
          "title": "",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Styled(styled.h2)>
          Schedule
        </Styled(styled.h2)>,
        <ul>
          <ScheduleEntry
            entry={
              Object {
                "id": 0,
                "title": "",
              }
            }
          />
          <ScheduleEntry
            entry={
              Object {
                "id": 1,
                "title": "",
              }
            }
          />
          <ScheduleEntry
            entry={
              Object {
                "id": 2,
                "title": "",
              }
            }
          />
        </ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Schedule",
        },
        "ref": null,
        "rendered": "Schedule",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ScheduleEntry
              entry={
                Object {
                  "id": 0,
                  "title": "",
                }
              }
            />,
            <ScheduleEntry
              entry={
                Object {
                  "id": 1,
                  "title": "",
                }
              }
            />,
            <ScheduleEntry
              entry={
                Object {
                  "id": 2,
                  "title": "",
                }
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "entry": Object {
                "id": 0,
                "title": "",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "entry": Object {
                "id": 1,
                "title": "",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "entry": Object {
                "id": 2,
                "title": "",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "ul",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Styled(styled.h2)>
            Schedule
          </Styled(styled.h2)>,
          <ul>
            <ScheduleEntry
              entry={
                Object {
                  "id": 0,
                  "title": "",
                }
              }
            />
            <ScheduleEntry
              entry={
                Object {
                  "id": 1,
                  "title": "",
                }
              }
            />
            <ScheduleEntry
              entry={
                Object {
                  "id": 2,
                  "title": "",
                }
              }
            />
          </ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Schedule",
          },
          "ref": null,
          "rendered": "Schedule",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ScheduleEntry
                entry={
                  Object {
                    "id": 0,
                    "title": "",
                  }
                }
              />,
              <ScheduleEntry
                entry={
                  Object {
                    "id": 1,
                    "title": "",
                  }
                }
              />,
              <ScheduleEntry
                entry={
                  Object {
                    "id": 2,
                    "title": "",
                  }
                }
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "entry": Object {
                  "id": 0,
                  "title": "",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "entry": Object {
                  "id": 1,
                  "title": "",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "function",
              "props": Object {
                "entry": Object {
                  "id": 2,
                  "title": "",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "ul",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
